src/main/java/com/google/cloud/tools/ide/login/AccountRoster.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.ide.login;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Internal class instantiated as a singleton member of {@link GoogleLoginState}. Manages a list
+ * of currently logged-in accounts. The first account in the list, if exists, is always the
```

Comment:
I agree that abstraction is better than this kind of implementation-based rule. However, there is one benefit in our case, which provides a simple and native solution to a problem:

My thought was to align with the order that UI will want and to avoid any re-ordering of the list when returning an account list to UI. For example, UI will want to show the active account at the top, and whenever a user switches an active account, the UI will want to put the next active account at the top while retaining the current order of other accounts. That is, the UI will want to show the account in the reserve order based on recency.

By managing the order internally here, UI can simply refresh the account following the order returned from `listAccounts()` as-is.